spyder/app/mainwindow.py

```diff
@@ -344,6 +344,25 @@ def last_plugin(self):
         # spyder/plugins/base::_switch_to_plugin
         return self.layouts.get_last_plugin()
 
+    def maximize_dockwidget(self, restore=False):
+        """
+        Maximize current dockwidget or restore it.
+
+        Parameters
+        ----------
+        restore : bool, optional
+            If the current dockwidget needs to be restored to its unmaximized
+            state. The default is False.
+
+        Returns
+        -------
+        None.
+
+        """
+        # Needed to prevent errors with the old API at
+        # spyder/plugins/base::_switch_to_plugin. See #spyder-ide/spyder#15164

```

Comment:
Please move this comment to be the docstring of this method and remove the one you added for it.